src/Engine/GameAction.java

```diff
@@ -30,6 +30,7 @@
    * Returns a GameEventQueue with the events that make up this action. If the action
    * was constructed incorrectly, this should return an empty GameEventQueue.
    */
+  public abstract Unit getActor();
```

Comment:
Why didn't you just add a default implementation?

Reply:
I dum

Reply:
Not necessarily. Having to provide an implementation means you have to think about what it means for each subclass, which might help avoid surprises down the line.
I'm generally in favor of terseness, but letting the compiler help you keep track of stuff can also help.